src/annotation-engine/utils.ts

```diff
@@ -21,70 +22,142 @@ export const drawRoundRect = (
+    path.quadraticCurveTo(
         coordinates.x + width,
         coordinates.y,
         coordinates.x + width,
         coordinates.y + radius,
     );
-    renderingContext.lineTo(coordinates.x + width, coordinates.y + height);
-    renderingContext.quadraticCurveTo(
+    path.lineTo(coordinates.x + width, coordinates.y + height);
+    path.quadraticCurveTo(
         coordinates.x + width,
         coordinates.y + height,
         coordinates.x + width,
         coordinates.y + height,
     );
-    renderingContext.lineTo(coordinates.x, coordinates.y + height);
-    renderingContext.quadraticCurveTo(coordinates.x, coordinates.y + height, coordinates.x, coordinates.y + height);
-    renderingContext.lineTo(coordinates.x, coordinates.y + radius);
-    renderingContext.quadraticCurveTo(coordinates.x, coordinates.y, coordinates.x + radius, coordinates.y);
-    renderingContext.fill();
-    renderingContext.closePath();
+    path.lineTo(coordinates.x, coordinates.y + height);
+    path.quadraticCurveTo(coordinates.x, coordinates.y + height, coordinates.x, coordinates.y + height);
+    path.lineTo(coordinates.x, coordinates.y + radius);
+    path.quadraticCurveTo(coordinates.x, coordinates.y, coordinates.x + radius, coordinates.y);
+    renderingContext.fill(path);
+
+    return path;
 };
 
-const drawLabel = (renderingContext: CanvasRenderingContext2D, label: string, from: Coordinates, to: Coordinates) => {
+const overloadStyle = (style: AnnotationStyle, customStyle?: PartialAnnotationStyle): AnnotationStyle => {
+    if(!customStyle) {
+        return style;
+    }
+    
+    // @Dimitri ou Manu : ici, le typage est assez lâche, et je n'ai pas encore réussi à le rendre plus strict.
+    // Comme on est dans un environnement contrôlé, est-ce que ça vaut le coup d'y passer plus de temps ?
+    const isObject = (item: any) => (item && typeof item === 'object' && !Array.isArray(item))
+
+    const mergeDeep = (target: any, source: any) => {
+        const output = {...target};
+        if (isObject(target) && isObject(source)) {
+            Object.keys(source).forEach(key => {
+                if (isObject(source[key])) {
+                    if (!(key in target))
+                        Object.assign(output, { [key]: source[key] });
+                    else {
+                        output[key] = mergeDeep(target[key], source[key]);
+                    }
+                } else {
+                    Object.assign(output, { [key]: source[key] });
+                }
+            });
+        }
+        
+        return output;
+    }
+
+    return mergeDeep(style, customStyle) as AnnotationStyle;
+}
+
+// @Dimitri et Manu l'orientation du label est désactivée en raison de la gestion du tracé avec des Path2D
+// faut-il trouver un moyen de le remettre en l'état, ou en profiter
+// pour corriger le comportement : toujours garder le label horizontal, mais en décalé "intelligent"
+// par rapport à la forme ?
```

Comment:
si on ne réimplémente pas la rotation, il faut à mon avis à minima deux choses (que tu as peut-être déjà implémenté):
 - centrer le label au milieu de la forme (verticalement ET/OU horizontalement)
 - avoir un label avec tous les bords ronds

Reply:
Je pense me diriger vers une solution de centrage.
Bien noté pour les 4 coins arrondis
Pour le positionnement de l'étiquette, je pense qu'il faudrait : 
- pour les points, le laisser où c'est
- pour les formes à plusieurs points : le centrer juste au-dessus du point le plus haut

De cette manière, on a un label qui ne se superpose jamais à la forme et qui est toujours proche d'un point de la forme => association visuelle implicite. En revanche, le label est tantôt à droite, tantôt à gauche, en particulier sur une ligne, selon que le point le plus haut est à droite ou à gauche...

Qu'en penses-tu ?

Reply:
il faudrait l'avis d'un UX sur le sujet.
tu peux implémenter la solution la plus simple et il faudra en discuter avec Antonin pendant la review (voir si on laisse en l'état ou si il faut améliorer)

Reply:
ok je vais partir comme ça !

Reply:
pour le moment j'ai mis en place le fait de placer le label au dessus du point le plus haut, et effectivement on verra avec un UX et le client :)

Reply:
bords ronds:
done !